squid/assets/logs/squid.yaml

```diff
@@ -79,37 +177,69 @@ pipeline:
         - 2019/10/11 09:14:49 kid1| Accepting HTTP Socket connections at local=[::]:3128 remote=[::] FD 11 flags=9
+        - 1725599181.650      0 10.212.128.16 TCP_MISS_ABORTED/000 0 - error:transaction-end-before-headers - HIER_NONE/- -
+        - 1725599185.240      0 10.212.128.16 TCP_DENIED/407 4660 GET http://clientservices.googleapis.com/chrome-variations/seed? - HIER_NONE/- text/html [http_request_headers] [http_response_headers]
+        - 10.212.128.16 - - [06/Sep/2024:10:32:43 +0530] "CONNECT fcmconnection.googleapis.com:443 HTTP/1.1" 407 4053 TCP_DENIED:HIER_NONE
+        - 10.212.128.16 - - [06/Sep/2024:10:34:44 +0530] "CONNECT mobile.events.data.microsoft.com:443 HTTP/1.1" 407 4277 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Code/1.92.2 Chrome/124.0.6367.243 Electron/30.1.2 Safari/537.36" TCP_DENIED:HIER_NONE
+        - 2019/10/11 09:14:49 kid1| Accepting HTTP Socket connections at local=[::]:3128 remote=[::] FD 11 flags=9
       grok:
-        matchRules: |
-          squid_access_parser %{number:timestamp:scale(1000)}\s+%{number:duration:scale(1000000)} %{ip:network.client.ip} %{regex("[^_]*"):network.protocol}_%{regex("[^/]*"):squid.status}/(000|%{integer:http.status_code}) %{number:network.bytes_written} %{word:http.method} (%{regex("(http|https)://[^\\s]+"):http.url}|%{notSpace:http.url_details.path}) (-|%{notSpace:squid.user_identity}) %{word:squid.peer_status}/(-|%{ip:network.destination.ip}) (:|-|%{notSpace:http.resource.content_type})?( \[%{regex("[^]]*"):http.headers.request}\])?( \[%{regex("[^]]*"):http.headers.response}\])?.*
+        supportRules: >
+          parse_prefix_combined_log_rule (-|%{ip:network.client.ip}) -
+          (-|%{word:squid.user_identity}) \[%{date("dd/MMM/yyyy:HH:mm:ss
+          Z"):timestamp}\] \"(-|%{word:http.method})
+          (-|%{regex("(http|https)://[^\\s]+"):http.url}|error:%{greedyData:error.message}|%{parse_url})
+          %{regex("[^\"]*"):http.version}\" (-|%{number:http.status_code})
+          %{number:network.bytes_written}
+
+          parse_suffix_combined_log_rule (NONE|%{regex("[^_]*"):network.protocol}_%{regex("[^:]*"):squid.status})\:(-|%{word:squid.peer_status}).*
+
+          parse_url %{hostname:http.url_details.host}(:%{number:http.url_details.port})?(%{regex("[^?|^#]*"):http.url_details.path})?(\?%{regex("[^#]*"):http.url_details.queryString:keyvalue("=","+%","","&")})?(\#%{notSpace:http.url_details.hash})?
+        matchRules: >
+          squid_access_parser
+          %{number:timestamp:scale(1000)}\s+(-|%{number:duration:scale(1000000)})
+          (-|%{ip:network.client.ip})
+          %{regex("[^_]*"):network.protocol}_%{regex("[^/]*"):squid.status}/(000|%{integer:http.status_code})
+          (-|%{number:network.bytes_written}) (-|%{word:http.method})
+          (-|%{regex("(http|https)://[^\\s]+"):http.url}|error:%{greedyData:error.message}|%{parse_url})
```

Comment:
I am not super clear why you are using `greedyData` here. It will try to take all Data by default and if the error is a small part of the log, it might lead to quite unperforming grok.

Unless you have a strong reason to do so, I will recommend using either `data` or even `notSpace` in case the error does include space here.

(same for line 190)

Reply:
Hey, @thibaultkrebs Thanks Will use data here.

Reply:
Hey, @thibaultkrebs the requested change is done.